### PR TITLE
add launch mp distributed job py module test=develop

### PR DESCRIPTION
**Pull request by Yancey0623**
We can launch nccl2 distributed training job with MP mode by:
``` bash
> python -m paddle.distributed.launch --gpus 8 train.py --update_method nccl2
```

### CODE REVIEW

File: python/paddle/fluid/distributed/launch.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
+# Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
```
Comment (typhoonzero) [inline python/paddle/fluid/distributed/launch.py]: can delete same file under tools

Comment (typhoonzero) [reply python/paddle/fluid/distributed/launch.py]: better to make usage like: 

`python -m fluid.distributed.launch train.py --your args`

Comment (Yancey0623) [reply python/paddle/fluid/distributed/launch.py]: done.

File: python/paddle/fluid/__init__.py [hunk 2]
```diff
@@ -160,10 +160,9 @@ def __bootstrap__():
             'sync_nccl_allreduce', 'limit_of_tmp_allocation',
             'times_excess_than_required_tmp_allocation'
         ]
-
-    core.init_gflags([sys.argv[0]] +
-                     ["--tryfromenv=" + ",".join(read_env_flags)])
-    core.init_glog(sys.argv[0])
+    program = sys.argv[0] if sys.argv[0].endswith(".py") else ""
```
Comment (typhoonzero) [inline python/paddle/fluid/__init__.py]: Not all train program endswith `.py` should find a way to identify when calling with `-m`

**Review by typhoonzero — APPROVED** [review]

**Pull request merged**